pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/PmdRuleCataloger.java

```diff
@@ -103,6 +104,11 @@ private void processCategoryFile(String language, String path) {
     // STEP 1: Turn the category file's XML into a Document object with a Root Element that we can actually use.
     Document doc = XmlReader.getInstance().getDocumentFromPath(path);
     Element root = doc.getDocumentElement();
+    // If the root node isn't of type 'ruleset', this isn't a valid category file, so we should just log something and skip it.
+    if (!root.getTagName().equalsIgnoreCase("ruleset")) {
+      SfdxMessager.getInstance().uxWarn("Skipping File " + path + ", which is not a valid Category definition file.", true);
```

Comment:
This needs to support language translations, doesn't it?

Also, the user may not know this is PMD related when they see this message.  Should probably associate it somehow with the engine visually.  Maybe a 'PMD Engine:' prefix to the message, etc.

Reply:
It does need to support translations. I'm not sure how that's implemented in Java.
Also, I agree that the message needs some work. As is, the message will be `Skipping file category/apex/somefile.xml, which is not a valid Category definition file.`. In addition to not being clearly related to PMD, it's also not clearly indicating which JAR that file's actually located inside of, so it could be hard to find it. If I can't find an easy way to identify the JAR, I might just scrap the logging altogether.

Reply:
How about a code that has the actual message mapped in the plugin's messages/*.json. Will support translation easily.

Here's what I'm imagining:

enum Code - would contain list of allowed message codes and their key name in plugin's messages/*.json.

For each message to display, `SfdxMessager.java` would hold:
Code code - code that can be mapped to a key in messages/*.json
String[] variables - variables in correct order that will be applied to replace parameters in messages

`PmdCatalogWrapper.ts` would get the code and variables from stdout, and then fetch corresponding message from messages/*.json. If it has a warning tag, it would display only when verbose is enabled. If it has an error tag, it would display every time.

I know this seems like expanding the logic further, but I feel SfdxMessager.java has a lot of potential in getting Java layer to give user-friendly and translated messages as well.

Reply:
Interesting idea. I'll give it a whirl.